### ISSUE TITLE
Unable to install packages when `--private` option is set

### ISSUE THREAD

**Issue by onema** [novemberfiveco/s3pypi#34]
I started using the `--private` option to submit my packages:
``` bash
AWS_PROFILE=production s3pypi --bucket pypi.my-company.io --force --private
```

But every time I did that, when installing the package I got the following message:
```bash
pip install my-package --extra-index-url https://pypi.my-company.io
Collecting my-package
  Could not find a version that satisfies the requirement my-package (from versions: )
No matching distribution found for my-package
```

If I submit it without the `--private`, it installs correctly.
Is this a bug? or am I doing something wrong?

**Comment by mdwint** [comment]
This is expected behaviour. In order for CloudFront to access private objects in S3, you'll need to set up an origin access identity.

https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-restricting-access-to-s3.html

**Comment by brokenthumbs** [comment]
So, I set up an origin access identity, but am still failing to install any packages.

I might also have a different problem (in addition?) where I can't install packages whether they are uploaded as private or public.

**Comment by jmrr** [comment]
Hi @brokenthumbs are you still experiencing this issue? I too can access the package  using `--find-links` but I only get `Connection timed out` with `--extra-index-url`. I've added the ALIAS record as I have configured the DNS through Route53. Should I leave some time for the record to propagate?

**Comment by brokenthumbs** [comment]
Not sure, I haven't retried any solutions here recently.

**Comment by jmrr** [comment]
Thanks @brokenthumbs I've created #35 to try and shed some light on this strange issue.

**Comment by jbasko** [comment]
Aren't these issues because when you request example.com/django/ via cloudfront, it doesn't request actually example.com/django/index.html? (which it does if you do it via s3)
I'm surprised that some here and on #35  have been able to work around this without implementing something like this:
https://aws.amazon.com/blogs/compute/implementing-default-directory-indexes-in-amazon-s3-backed-amazon-cloudfront-origins-using-lambdaedge/

**Comment by mdwint** [comment]
This has been fixed in s3pypi 1.0.0. An Origin Access Identity is included in the Terraform configuration.
